### ISSUE TITLE
QDrawer breakpoint doesn't respect greater width once triggered

### ISSUE THREAD

**Issue by ImSeaWorld** [quasarframework/quasar#12811]
### What happened?

Shrink window to smaller than the breakpoint specified, open the q-drawer. Once you full screen again, it will keep the q-drawer in it's breakpoint state.

### What did you expect to happen?

Q-drawer reset to it's original state once larger than the breakpoint.
This: https://codepen.io/imseaworld/pen/JjMYxNN

### Reproduction URL

https://quasar.dev/
https://codepen.io/imseaworld/pen/dyJomor

### How to reproduce?

1. Go to provided repro link
2. Shrink window to lower than the breakpoint for the sidebar
3. Open sidebar
4. Full screen window or make inner width larger than specified breakpoint

### Flavour

Quasar CLI with Webpack (@quasar/cli | @quasar/app-webpack)

### Areas

Components (quasar)

### Platforms/Browsers

Chrome, Microsoft Edge

**Comment by rstoenescu** [comment]
Hi,

If it opens as mobile is stays as mobile until you close it. This is by design to not mess up drawer state completely.

**Comment by ImSeaWorld** [comment]
> This is by design to not mess up drawer state completely.

I don't see how it would do that since the functionality already exists, it's just not working as intended.

> If it opens as mobile is stays as mobile until you close it.

This does not happen. It will remain mobile till the breakpoint is activated again. I've fixed the issue of the button disappearing(like what happens on quasar.dev), but why would it reset itself after the breakpoint is activated again to go back to desktop?

![chrome_c6LpaJp1fD](https://user-images.githubusercontent.com/20188588/158714372-ca419df5-c2b4-46a1-b70f-5a82aba4958a.gif)

This isn't the functionality expected from what you've said. 

This would be the expected behavior: https://codepen.io/imseaworld/pen/JjMYxNN
Since you didn't look at the last reproduction I made, here's a gif:
![chrome_hOEBDBg0x7](https://user-images.githubusercontent.com/20188588/158717092-2c05d802-827f-47c8-9a8e-8180e19914ab.gif)

**Comment by ImSeaWorld** [comment]
@rstoenescu I could understand how my issue could have been misinterpreted, but this is a REAL edge case that's very obvious on the documentation. 

![chrome_NNLYzuaE2I](https://user-images.githubusercontent.com/20188588/158934150-e22554b2-8dba-4140-83e3-1a45b00e5086.gif)

It's not detecting when the width grows after the breakpoint has been triggered, only when the window is shrinking again. Very easy to overlook but this is something I noticed on the documentation(which is fantastic btw) when I first found Quasar. If this is a feature, then I'm not going to trust `breakpoint` properties anymore.
